### PR TITLE
Remove scheduled backup Lambda

### PR DESCRIPTION
**Pull request by jacobwinch**
## What does this change?

This PR removes the `MembershipSub-Promotions-Scheduled-Dynamo-Backup` Lambda. We now have nightly backups via https://github.com/guardian/aws-backup, so I think this bespoke solution is now superfluous. There are more details on the new backup tooling [here](https://docs.google.com/document/d/11EZtuVHCnjavE9AYLroiuDUsMzbp2ulFBtj4VkyCKNc/edit#heading=h.vwt7syo8ng40).

## How to test

This Lambda is `PROD` only so it's tricky to test this - I think we'll need to deploy this branch to `PROD`.

## How can we measure success?

There will be less JavaScript and infrastructure to maintain! 

The new backups are also more secure (it is not possible for a Guardian employee to delete them accidentally and an attacker could not destroy them if they compromised the account) because they are protected by [vault lock](https://docs.aws.amazon.com/aws-backup/latest/devguide/vault-lock.html).

## Have we considered potential risks?

I don't think there are any particular risks associated with this as these tables are now backed up by a department-wide solution: https://github.com/guardian/aws-backup.

## TODO

- [x] Double check that we now have 14 days of backups for these tables in AWS Backup
- [ ] Delete all old DynamoDB-style backups[^1] manually (the Lambda that is being deleted currently deletes backups older than 14-days, so once this has gone we need to clean up manually)

[^1]: The new tooling creates slightly different backups via the AWS Backup service and these will not be deleted

### CODE REVIEW

File: lambdas/src/MembershipSub-Promotions-Scheduled-Dynamo-Backup.js [hunk 1]
```diff
@@ -1,64 +0,0 @@
...
-                    return `Successfully${latestBackupARN ? ' created new backup and' : ''} deleted ${results.length} backups for table: ${TableName}`;
-                })
-            });
-    };
-}
-
-export const handler = (event, context, callback) => {
-
-    const TOUCHPOINT_BACKEND = /PROD$/.test(context.functionName) ? 'PROD' : 'CODE';
-    const RETENTION_DAYS = 14; // a number < 1 will delete everything except the backup `createTodaysBackup` just created
```
Comment (jacobwinch) [inline lambdas/src/MembershipSub-Promotions-Scheduled-Dynamo-Backup.js:52]: The new solution is using the same retention period: https://github.com/guardian/aws-backup/blob/12ccfc4c4cd20c5eebfa2803beaee8521856b86f/cdk/lib/aws-backup.ts#L34

File: .github/workflows/memsub-promotions-lambdas.yml [hunk 2]
```diff
@@ -81,7 +81,8 @@ jobs:
                   functionNames:
                     - MembershipSub-Promotions-to-PromoCode-View-
                     - MembershipSub-PromoCode-View-Dynamo-to-Salesforce-
-                    - MembershipSub-Promotions-Scheduled-Dynamo-Backup-
+                dependencies:
```
Comment (jacobwinch) [inline .github/workflows/memsub-promotions-lambdas.yml:84]: I noticed that this was missing; I don't think it's strictly necessary for this PR to work.

**Review by paulbrown1982 — APPROVED** [review]
👍 thank you for doing this!

**Comment by jacobwinch** [comment]
> 👍 thank you for doing this!

Thanks @paulbrown1982 - I'll deploy this next week just in case I've missed something!

**Comment by jacobwinch** [comment]
I've [deployed this successfully](https://riffraff.gutools.co.uk/deployment/view/c98304ec-3b1c-4bd9-b68f-24e03a284103) and can see that the Lambda has been deleted:

```
[13:33:52] ScheduledBackupLambdaPROD (AWS::Lambda::Function): DELETE_IN_PROGRESS
[13:33:57] ScheduledBackupLambdaPROD (AWS::Lambda::Function): DELETE_COMPLETE
```

**Comment by jacobwinch** [comment]
This Lambda was creating backups with a [`Scheduled-Backup` suffix](https://github.com/guardian/memsub-promotions/blob/938bd3a2c8ccf67d173f98154dda383e69c3d968/lambdas/src/MembershipSub-Promotions-Scheduled-Dynamo-Backup.js#L7):

image

These have now been deleted manually:

image

**Pull request merged**